src/Core/Misc/ExtensionMethods.cs

```diff
@@ -59,12 +64,11 @@ public static bool IsControlOrContent(this A11yElement e)
         /// <returns>UIA_ClassNamePropertyId</returns>
         public static string GetClassName(this A11yElement e)
         {
-            if (e.Properties != null && e.Properties.ContainsKey(PropertyType.UIA_ClassNamePropertyId))
-            {
-                return e.Properties[PropertyType.UIA_ClassNamePropertyId].TextValue;
-            }
-            else
-                return null;
+            if (e == null) throw new ArgumentNullException(nameof(e));
+            if (e.Properties == null) return null;
+            if (!e.Properties.ContainsKey(PropertyType.UIA_ClassNamePropertyId)) return null;
```

Comment:
Great place for a TryGetValue:

if (e.Properties.TryGetvalue(PropertyType.UIA_ClassNamePropertyId, out A11yProperty property))
    return property.TextValue;

return null;

Reply:
I don't disagree. However, I have _tried_ to draw the line with these changes to items found by the analyzers.

Reply:
Makes sense.

---
In reply to: [311284107](https://github.com/microsoft/axe-windows/pull/130#discussion_r311284107) [](ancestors = 311284107)